packages/react-composites/src/composites/MeetingComposite/MeetingComposite.tsx

```diff
@@ -32,16 +33,30 @@ export type MeetingCompositeProps = {
  * Meeting Composite brings together key components to provide a full meeting experience out of the box.
  */
 export const MeetingComposite = (props: MeetingCompositeProps): JSX.Element => {
-  const { callAdapter, chatAdapter, fluentTheme } = props;
+  const { meetingAdapter, fluentTheme } = props;
+
+  if (!meetingAdapter) {
+    throw 'Meeting adapter is undefined';
+  }
+
+  const [callAdapter, setCallAdapter] = useState<CallAdapter>(new MeetingBackedCallAdapter(meetingAdapter));
+  const [chatAdapter, setChatAdapter] = useState<ChatAdapter>(new MeetingBackedChatAdapter(meetingAdapter));
+
+  useEffect(() => {
```

Comment:
meetingAdapter is a `prop` that is never changed internally. So, you should be able to simply `useRef` for `callAdapter` and `chatAdapter` and directly create those two (without a `useEffect` block).

`useEffect` is intended to respond to some external change that isn't a directly result of a props change.

Reply:
Ah, I guess the idea is to not recreate the adapters if props other than meetingAdapter change?

Even in that case, I'd just store them all as ref (because it is never the case that any of them change without a prop change):


    const { meetingAdapter: newMeetingAdapter, fluentTheme } = props;
    const meetingAdapter = useRef(newMeetingAdapter);
    const chatAdapter = useRef(new MeetingBackedCallAdapter(meetingAdapter));
    ...

    // On prop update only.
    if (newMeetingAdapter !== meetingAdapter.current) {
        meetingAdapter.current = newMeetingAdatper;
        chatAdatper.current = new MeetingBackedChatAdapter(newMeetingAdapter);
        ...
    }

Reply:
Updated to use `useMemo` instead to make it much simpler:
```
const callAdapter: CallAdapter = useMemo(() => new MeetingBackedCallAdapter(meetingAdapter), [meetingAdapter]);
const chatAdapter: ChatAdapter = useMemo(() => new MeetingBackedChatAdapter(meetingAdapter), [meetingAdapter]);
```